unime/src/routes/(app)/scan/+page@.svelte

```diff
@@ -189,111 +117,38 @@
   onMount(async () => {
     console.log('onMount: /scan');
     document.documentElement.querySelector('body')!!.classList.add('transparent');
-    // permissionsGiven = await checkScanPrerequisites();
 
     // TODO find a good way to test if not dev_mode. This will have to be checked after $state is loaded.
     startScan();
   });
 </script>
 
-<!-- <main
-  class="transition-colors-250 grid flex-1 grid-rows-[2fr_auto] bg-orange-300 transition-transform"
-  class:transparent={isMobile}
->
-  <div class="flex flex-col items-center p-6">
-    <h1 class="p-6 text-3xl font-medium">{selected.label}</h1>
-    <div class="overflow-y-auto">
-      <div class="mr-2">
-        <svelte:component this={selected.component} {onMessage} />
-      </div>
-    </div>
-  </div>
-</main> -->
-
-<!-- ############ -->
-<!-- <div class="h-screen">
-  <p class="absolute m-4 w-fit rounded-full bg-slate-600 px-4 py-2 text-slate-200">
-    scanning: {scanning}
-  </p>
-
-  {#if !scanning}
-    <div class="h-screen bg-orange-300" class:invisible={scanning}>
-      <Button variant="secondary" on:click={startScan}>SCAN</Button>
-    </div>
-  {:else}
-
-  <div class="h-screen bg-gradient-to-r from-violet-600 via-green-400 to-violet-500">
-        <div class="">
-            <p>Aim your camera at a QR code</p>
-        </div>
-        <div class="absolute top-1/2 left-1/2">
-            <div class="w-20 h-20 bg-slate-300 rounded overflow-hidden"></div>
-        </div>
-    </div>
-  {/if}
-</div> -->
-
-<!-- {#if scanning}
-    scanning
-    <div class="h-screen bg-gradient-to-br from-violet-600 to-transparent" />
-  {/if} -->
-
-<!-- scanning in progress -->
-<!-- <div
-    class="h-screen bg-gradient-to-br from-violet-600 to-transparent"
-    class:invisible={!scanning}
-  />
-  <div class="h-full w-full">test123</div> -->
-
-<!-- ############ -->
-
 <div class="content-height flex flex-col items-stretch">
-  <!-- <div class="flex h-screen flex-col items-stretch"> -->
   <div class="hide-scrollbar grow overflow-x-hidden overflow-y-scroll">
     <div class="flex h-full w-full flex-col">
       <!-- visible when NOT scanning -->
       <div
         class:invisible={scanning}
-        class="relative flex h-full flex-col items-center justify-center bg-silver p-8 dark:bg-navy"
+        class="relative flex h-full flex-col items-center justify-center space-y-4 bg-silver p-8 dark:bg-navy"
       >
-        {#if permissions === 'denied'}
-          <div class="flex flex-col items-center space-y-4">
-            <div class="rounded-lg bg-rose-100 px-8 py-4 text-rose-500">
-              {$LL.SCAN.NO_PERMISSION_1()}<br />{$LL.SCAN.NO_PERMISSION_2()}
+        {#if permissions_nullable !== 'granted'}
```

Comment:
Although this solution seems to work, now every time I go to the `Scan` tab, for a brief moment the "No permissions to access the camera" prompt is visible before opening the camera. So it registers a bit late that the permission is actually granted. 

I remember this being an issue before but I am not completely sure.. 

Is there some way of changing the logic that only when failing to start the scanner the prompt is shown?

Reply:
I believe that when the permissions are not loaded yet (but are `granted`), it would still briefly show the "no permissons given" due it `!== 'granted'` at the time of check. That's why I added a check against `null` and it seems to work now.